scripts/build-docs.js

```diff
@@ -0,0 +1,158 @@
+        });
+    }
+
+    if (targetFormat === KRAMDOWN_FORMAT) {
+        // Kramdown uses ~~~ for code blocks
+        result = result.replace(/```/g, '~~~');
+    }
+
+    return result;
+}
+
+/**
+ * Builds markdown docs suitable for ingestion into the website generator
+ */
+function buildDocs() {
+    let jsdoc2md = require('jsdoc-to-markdown');
+    let docSrcDirPath = path.resolve('doc');
+    let docDestDirPath = path.resolve(DEST_DIR, 'docs');
+
+    // Create docs dest dir, if not already present
+    mkdirp.sync(docDestDirPath);
+
+    let docPromises = [];
+
+    // Initiate index copy - Add to promise array for concurrency
+    docPromises.push(
+        fsThen.readFile(path.join(docSrcDirPath, 'index.md'), {encoding: 'UTF-8'}).then(buffer => {
+            buffer = buildDocHeader('Client App SDK') + buffer;
+
+            buffer = transformGithubMarkdown(buffer, docMdOutputFormat);
+
+            buffer = transformRelativeLinks(buffer, relativeLinkExtension);
+
+            return fsThen.writeFile(path.join(docDestDirPath, 'index.md'), buffer, {encoding: 'UTF-8'});
+        })
+    );
+
+    // Generate Class docs from jsdoc in src
+    let partialsPattern = path.join(docSrcDirPath, 'partials', '**', '*.hbs');
+    let helpers = path.join(docSrcDirPath, 'helpers.js');
+
+    return jsdoc2md.getTemplateData({
+        cache: null,
+        files: 'src/**/*.js'
+    }).then(templateData => {
+        templateData.forEach(currIdentifier => {
+            if (!currIdentifier || currIdentifier.kind !== 'class') {
+                return;
+            }
+
+            let {name: className} = currIdentifier;
+            if (className) {
+                // Remove the module as the parent of the classes for the purposes of doc generation
+                currIdentifier.scope = 'global';
+                currIdentifier.memberof = '';
+
+                let template = `{{#class name="${className}"}}{{>docs}}{{/class}}`;
+
+                docPromises.push(
+                    jsdoc2md.render({
+                        data: templateData,
+                        template: template,
+                        partial: partialsPattern,
+                        helper: helpers,
+                        'heading-depth': 2,
+                        'example-lang': 'js',
+                        purecloudCustom: {
+                            linkExtension: relativeLinkExtension
+                        }
+                    }).then(output => {
+                        output = buildDocHeader(className) + output;
+
+                        output = transformGithubMarkdown(output, docMdOutputFormat);
+
+                        return fsThen.writeFile(path.join(docDestDirPath, `${className}.md`), output);
+                    })
+                );
+            }
+        });
+
+        return Promise.all(docPromises);
+    }).catch(reason => {
+        let errMsg = 'Documentation generation failed';
+        console.error(errMsg, reason);
```

Comment:
i think we could reject here with an object.  maybe something like {msg, reason}.  This lets the caller deal with how to handle ... logging (and which logger to use) and/or process.exit.

This is related to the cli handler comment above.

Reply:
I think it makes sense then to not catch the error at all, and just leave it up to the caller to handle the error, right?

Reply:
yeah, i think that makes sense here since we'll mostly just catch and log it anyway.  if it was an api, maybe we'd come up with a few types of errors (enum) and throw that and the nitty gritty deets.

mostly this will just get rid of a log statement here in case it is handled by the caller and the log would be a red herring.